server/routes/induction/common/workedBeforeController.ts

```diff
@@ -18,7 +18,7 @@ export default abstract class WorkedBeforeController extends InductionController
     this.addCurrentPageToFlowHistoryWhenComingFromCheckYourAnswers(req)
 
     // Check if we are in the midst of changing the main induction question set (in this case from short route to long route)
-    if (req.session.updateInductionQuestionSet) {
+    if (req.session.updateInductionQuestionSet || req.session.pageFlowHistory) {
```

Comment:
Whilst the bug was observed on the "previous work experience types" page, the fix to it is in the superclass controller for the previous page ("have they worked before?" page)

For the Back links to work correctly and know what page to link back to, we need the previous page being added to the page flow history. Hence this change in the previous page's controller 👍